ocaml/xapi/certificates_sync.ml

```diff
@@ -97,3 +97,53 @@ let update ~__context =
   let* () = sync ~__context ~type':`host in
   let* () = sync ~__context ~type':`host_internal in
   Ok ()
+
+let internal_error fmt =
+  fmt
+  |> Printf.kprintf @@ fun msg ->
+     error "%s" msg ;
+     raise Api_errors.(Server_error (internal_error, [msg]))
+
+let path ~__context cert =
+  let uuid = Db.Certificate.get_uuid ~__context ~self:cert in
+  let type' = Db.Certificate.get_type ~__context ~self:cert in
+  let prefix =
+    match type' with
+    | `host ->
+        !Xapi_globs.trusted_certs_dir
```

Comment:
The trusted certs dir is used by `ca certs, not `host ones. The correct filename for the certificate can be build by concatenating it with the field `name` in their record.

Reply:
We use the following paths to build bundles which we pass to stunnel for client connections:

```
regen_bundle "/etc/stunnel/certs"      "/etc/stunnel/xapi-stunnel-ca-bundle.pem"
regen_bundle "/etc/stunnel/certs-pool" "/etc/stunnel/xapi-pool-ca-bundle.pem"
```
These correspond to these names:

```
let trusted_certs_dir = ref "/etc/stunnel/certs"
let trusted_pool_certs_dir = ref "/etc/stunnel/certs-pool"
```

So where are the certs that belong to a host in the pool that need to be removed and which are tracked in the database as host and host-internal certificates? According to `xe` they don't have a meaningful name, which could be a different problem:

```

Reply:
The correct paths are:
```
Xapi_globs.server_cert_path
Xapi_globs.server_cert_internal_path
```

```